packages/salesforcedx-vscode-core/src/commands/forceSourceDiff.ts

```diff
@@ -5,130 +5,88 @@
  * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
 
-import {
-  CliCommandExecutor,
-  Command,
-  DiffResultParser,
-  SfdxCommandBuilder
-} from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
-import { ContinueResponse } from '@salesforce/salesforcedx-utils-vscode/out/src/types';
+import { SourceComponent } from '@salesforce/source-deploy-retrieve';
+import * as path from 'path';
 import * as vscode from 'vscode';
 import { channelService } from '../channels';
+import {
+  MetadataCacheExecutor,
+  MetadataCacheResult
+} from '../conflict/metadataCacheService';
+import { workspaceContext } from '../context';
 import { nls } from '../messages';
-import { notificationService, ProgressNotification } from '../notifications';
-import { taskViewService } from '../statuses';
+import { notificationService } from '../notifications';
 import { telemetryService } from '../telemetry';
-import { getRootWorkspacePath, hasRootWorkspace, OrgAuthInfo } from '../util';
-import {
-  FilePathGatherer,
-  SfdxCommandlet,
-  SfdxCommandletExecutor,
-  SfdxWorkspaceChecker
-} from './util';
-
-export class ForceSourceDiffExecutor extends SfdxCommandletExecutor<string> {
-  public build(filePath: string): Command {
-    const commandBuilder = new SfdxCommandBuilder()
-      .withDescription(nls.localize('force_source_diff_text'))
-      .withArg('force:source:diff')
-      .withLogName('force_source_diff')
-      .withFlag('--sourcepath', filePath)
-      .withJson();
-    return commandBuilder.build();
-  }
-
-  public async execute(response: ContinueResponse<string>): Promise<void> {
-    const startTime = process.hrtime();
-    const cancellationTokenSource = new vscode.CancellationTokenSource();
-    const cancellationToken = cancellationTokenSource.token;
-
-    const execution = new CliCommandExecutor(this.build(response.data), {
-      cwd: getRootWorkspacePath(),
-      env: { SFDX_JSON_TO_STDOUT: 'true' }
-    }).execute(cancellationToken);
+import { FilePathGatherer, SfdxCommandlet, SfdxWorkspaceChecker } from './util';
 
-    channelService.streamCommandStartStop(execution);
-
-    let stdOut = '';
-    execution.stdoutSubject.subscribe(realData => {
-      stdOut += realData.toString();
-    });
+const workspaceChecker = new SfdxWorkspaceChecker();
 
-    execution.processExitSubject.subscribe(async exitCode => {
-      this.logMetric(execution.command.logName, startTime);
-      await handleDiffResponse(exitCode, stdOut);
-    });
+/**
+ * Perform file diff and execute VS Code diff comand to show in UI.
+ * It matches the correspondent file in compoennt.
+ * @param localFile local file
+ * @param remoteComponent remote source component
+ * @returns {Promise<void>}
+ */
+async function diffFile(
```

Comment:
I refactored this to conflictDetectionService to keep things cleaner here. Let me know if that sounds good once we merge ours.

Reply:
yeah, that looks good